### DISCUSSION TITLE
brew doctor

### DISCUSSION THREAD

**Discussion by code202021** [Homebrew/discussions#207] [Everyday usage] [answered]
Hello,

I'm trying to use homebrew to install the latest version of python as well as pip so that I can create a vpn using algo vpn see https://github.com/trailofbits/algo

I am using macOS 11.0.1 Big Sur

I just want to make sure everything is setup correctly because there are some confusing things to me regarding the setup of python. For example you can have the default version of python for macOS and then if you can have python3 installed separately instead of doing an upgrade, but then you can link it to the default if I understand correctly.

After putting in the command

`$brew doctor`

in terminal I receive:

> Warning: "config" scripts exist outside your system or Homebrew directories.
> 
> `./configure` scripts often look for *-config scripts to determine if
> 
> software packages are installed, and which additional flags to use when
> compiling and linking.
> 
> Having additional scripts in your path can confuse software installed via
> Homebrew if the config script overrides a system or Homebrew-provided
> script of the same name. We found the following "config" scripts:
>   /Library/Frameworks/Python.framework/Versions/3.7/bin/python3.7-config
>   /Library/Frameworks/Python.framework/Versions/3.7/bin/python3.7m-config
>   /Library/Frameworks/Python.framework/Versions/3.7/bin/python3-config
> 
> Warning: You have unlinked kegs in your Cellar.
> Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
> those kegs to fail to run properly once built. Run `brew link` on these:
>   python@3.9
> 

I mainly just want to make sure I don't cause any instability with macOS and I want to be able to run the algo vpn setup. I am not trying to change the default version of python that macOS uses and I don't believe I need to have the python default python linked in any way with python3. If these warnings are valid can someone please help me fix them?

**Comment by SMillerDev** [comment, marked as answer]
These warnings will likely not interfere with any other setup. But it does mean that you have a python version from `python.org` installed next to the Homebrew one.

**Reply by code202021** [reply]
Thank you for your reply.

Would it be better if I removed one though? If yes, do you know of any links that talk about how to do that?

**Reply by SMillerDev** [reply]
https://xkcd.com/1987/ explains some of the problems with this kind of setup :smile: 

and https://www.techjunkie.com/macos-uninstall-python3/ explains how to remove the python from python.org

**Reply by code202021** [reply]
Thank you for your reply and the links 🙂

**Reply by code202021** [reply]
I received an error when trying to setup the algo vpn and I think it may be due to the two versions that I had of python3. I have Time Machine backups so I may go ahead and use one to try to get macOS back to where it was prior to the state it is now in. If I wanted to try to get some help in a forum without using Time Machine is there a forum that you would suggest?

**Reply by SMillerDev** [reply]
If you're trying to setup algo vpn you should check if they have a forum (they seem to https://github.com/trailofbits/algo/discussions). Or you could use https://apple.stackexchange.com/ for a more generic Apple forum.

**Reply by code202021** [reply]
Thank you 🙏
